### ISSUE TITLE
What if log_theta is parametrized by another network?

### ISSUE THREAD

**Issue by Xiaohui9607** [agadetsky/pytorch-pl-variance-reduction#1]
I am confused about the comments in estimators.py:
"in all computations (including those made by the user later), we don't
"want to backpropagate past "logits" into the model. We make a detached
"copy of logits and rebuild the graph from the detached copy to z

logits = logits.detach().require_grad(True)

if logits are parametrized by a network theta, then if you detach here then the network won't get updated right?

**Comment by agadetsky** [comment]
@Xiaohui9607 all will be fine if you further propagate gradient obtained by the estimator. See example at https://pydrobert-pytorch.readthedocs.io/en/latest/estimators.html
